common/ayon_common/distribution/addon_info.py

```diff
@@ -114,17 +131,26 @@ class DependencyItem(object):
     def from_dict(cls, package):
         sources = []
         unknown_sources = []
-        for source in package.get("sources", []):
+        package_sources = package.get("sources")
+        require_distribution = package_sources is not None
+        for source in (package_sources or []):
             source_type = source.get("type")
             if source_type == UrlType.FILESYSTEM.value:
-                source_addon = LocalAddonSource(
+                source_addon = LocalSourceInfo(
                     type=source_type, path=source["path"])
             elif source_type == UrlType.HTTP.value:
-                source_addon = WebAddonSource(
-                    type=source_type, url=source["url"])
+                url = source["path"]
+                source_addon = WebSourceInfo(
+                    type=source_type,
+                    url=url,
+                    headers=source.get("headers")
+                )
             elif source_type == UrlType.SERVER.value:
-                source_addon = ServerResourceSource(
-                    type=source_type, filename=source["filename"])
+                source_addon = ServerSourceInfo(
+                    type=source_type,
+                    filename=source.get("filename"),
+                    path=source.get("path")
+                )
             else:
                 print(f"Unknown source {source_type}")
                 unknown_sources.append(source)
```

Comment:
Seeing this being duplicated logic from above I think we can have a helper function to `convert_source(source)` or `convert_sources(sources)` if we want to keep track of the unknown ones directly too.